### ISSUE TITLE
Reader::createFromStream not reading php://input

### ISSUE THREAD

**Issue by nickma42** [thephpleague/csv#314]
### Bug Report

| Information | Description |
|--------------|---------|
| Version | 9.1.4 |
| PHP version | 7.2.9 |
| OS Platform | Ubuntu 18.04.1 LTS |

#### Summary

Trying to use `Reader::createFromStream` with `fopen('php://input', 'r')` as the stream resource results in no data being loaded by the reader.

#### Standalone code, or other way to reproduce the problem

```php
$csv = Reader::createFromStream(fopen('php://input', 'r'));
echo $csv->getContent();
```

(This unfortunately needs to be run within the context of an HTTP POST request so that `php://input` actually has some data - I wasn't sure if there was a way to simulate that for the purpose of this bug report)

#### Expected result

The echo should return the contents of the `php://input` stream as shown in this working example:

```php
$csv = Reader::createFromString(file_get_contents('php://input'));
echo $csv->getContent();
```

#### Actual result

Nothing returned from the echo.

**Comment by nyamsprod** [comment]
This is expected behavior as explain in PHP documentation for PHP I/O stream wrappers
see http://php.net/manual/en/wrappers.php.php#wrappers.php.input
TL;DR a `php://input` may be seekable but every time you rewind the stream you loose `php://input` content so **you are required to re-open the stream which the library will never do.**

So you can either use you solution with `file_get_contents` or use `stream_copy_to_stream` as shown below:

in `test.php`

```php
<?php

use League\Csv\Reader;

require 'autoload.php';

echo "==== using file_get_contents ====", PHP_EOL;
$data = file_get_contents('php://input');
$csv = Reader::createFromString($data);
var_dump(iterator_to_array($csv));

echo "==== fgetcsv before rewind ====", PHP_EOL;
$rsrc = fopen('php://input', 'r');
var_dump(stream_get_meta_data($rsrc));
while (false !== ($record = fgetcsv($rsrc))) {
	var_dump($record);
}
$res = rewind($rsrc);
var_dump($res);
echo "==== fgetcsv after rewind ====", PHP_EOL;
while (false !== ($record = fgetcsv($rsrc))) {
	var_dump($record);
}
fclose($rsrc);

echo "==== copy stream ====", PHP_EOL;
$tmp = tmpfile();
stream_copy_to_stream(fopen('php://input', 'r'), $tmp);
$data = Reader::createFromStream($tmp);
var_dump(iterator_to_array($csv));
fclose($tmp);
```

Then you can simply call this script using cURL with the following command line:

```bash
$ curl -d "foo,bar,baz" -X POST http://localhost/csv/test.php
```

**Comment by nickma42** [comment]
Understood - thanks for the quick reply and the great explanation :)
